plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProvider.java

```diff
@@ -107,7 +119,9 @@ public boolean isAvailable() {
         vfsLocation.setPath( vfsConnectionDetails.getType() + "://" + vfsConnectionDetails.getDomain() );
         vfsLocation.setDomain( vfsConnectionDetails.getDomain() );
         vfsLocation.setConnection( connectionDetails.getName() );
-        vfsTree.addChild( vfsLocation );
+        if ( connectionTypes.isEmpty() || connectionTypes.contains( connectionDetails.getType() ) ) {
```

Comment:
Technically because there's a load( null, null ) this can have a NPE. need a null check or change load to construct an empty list.

Reply:
Good catch, will update in another PR